src/Recurrence.ts

```diff
@@ -212,4 +207,145 @@ export class Recurrence {
+     * occurrence will happen on 2022-03-31. The reason being that February
+     * does not have 31 days, yet RRule sets `bymonthday` to `31` for lack of
+     * having a better alternative.
+     *
+     * In order to fix this, `after` will move into the past day by day. Each
+     * day, the next occurrence is checked to be after the given number of
+     * months or years. By moving `after` into the past day by day, it will
+     * eventually calculate the next occurrence based on `2022-01-28`, ending up
+     * in February as the user would expect.
+     */
+    private nextAfter(after: Moment, rrule: RRule): Date {
+        // We need to remove the timezone, as rrule does not regard timezones and always
+        // calculates in UTC.
+        // The timezone is added again before returning the next date.
+        after.utc(true);
+        let next = window.moment(rrule.after(after.toDate()));
+
+        // If this is a monthly recurrence, treat it special.
+        const monthMatch = this.toText().match(/every( \d+)? month(s)?(.*)?/);
+        if (monthMatch !== null) {
+            next = Recurrence.nextAfterMonths(after, next, rrule, monthMatch[1]);
+        }
+
+        // If this is a yearly recurrence, treat it special.
+        const yearMatch = this.toText().match(/every( \d+)? year(s)?(.*)?/);
+        if (yearMatch !== null) {
+            next = Recurrence.nextAfterYears(after, next, rrule, yearMatch[1]);
+        }
+
+        // Here we add the timezone again that we removed in the beginning of this method.
+        return Recurrence.addTimezone(next).toDate();
+    }
+
+    /**
+     * nextAfterMonths calculates the next date after `skippingMonths` months.
+     *
+     * `skippingMonths` defaults to `1` if undefined.
+     */
+    private static nextAfterMonths(
+        after: Moment,
+        next: Moment,
+        rrule: RRule,
+        skippingMonths: string | undefined,
+    ): Moment {
+        // Parse `skippingMonths`, if it exists.
+        let parsedSkippingMonths: Number = 1;
+        if (skippingMonths !== undefined) {
+            parsedSkippingMonths = Number.parseInt(skippingMonths.trim(), 10);
+        }
+
+        // While we skip the wrong number of months, move `after` one day into the past.
+        while (Recurrence.isSkippingTooManyMonths(after, next, parsedSkippingMonths)) {
+            // The next line alters `after` to be one day earlier.
+            // Then returns `next` based on that.
+            next = Recurrence.fromOneDayEarlier(after, rrule);
+        }
+
+        return next;
+    }
+
+    /**
+     * isSkippingTooManyMonths returns true if `next` is more than `skippingMonths` months after `after`.
+     */
+    private static isSkippingTooManyMonths(after: Moment, next: Moment, skippingMonths: Number): boolean {
+        let diff = next.month() - after.month();
+        if (diff < 0) {
+            diff = diff + 12;
```

Comment:
Does the number of years need to be taken in to account? Like what if diff was -13?

Reply:
I see from the later commit that the answer was yes... 😄

Reply:
Indeed! Found it thanks to your tests.